ns/nns.go

```diff
@@ -116,40 +83,21 @@ func (n *NNS) Dial(address string) error {
 //
 // See also https://docs.neo.org/docs/en-us/reference/nns.html.
 func (n *NNS) ResolveContainerName(name string) (cid.ID, error) {
-	res, err := n.neoClient.call(n.nnsContract, "resolve", []smartcontract.Parameter{
-		{
-			Type:  smartcontract.StringType,
-			Value: name + ".container",
-		},
-		{
-			Type:  smartcontract.IntegerType,
-			Value: big.NewInt(int64(nns.TXT)),
-		},
-	})
-	if err != nil {
-		return cid.ID{}, fmt.Errorf("invoke NNS contract: %w", err)
-	}
-
-	if res.State != vmstate.Halt.String() {
-		return cid.ID{}, fmt.Errorf("NNS contract fault exception: %s", res.FaultException)
-	} else if len(res.Stack) == 0 {
-		return cid.ID{}, errors.New("empty stack in invocation result")
-	}
-
-	itemArr, err := res.Stack[len(res.Stack)-1].Convert(stackitem.ArrayT) // top stack element is last in the array
+	item, err := unwrap.Item(n.invoker.Call(n.nnsContract, "resolve",
+		name+".container", int64(nns.TXT),
+	))
 	if err != nil {
-		return cid.ID{}, fmt.Errorf("convert stack item to %s", stackitem.ArrayT)
+		return cid.ID{}, fmt.Errorf("contract invocation: %w", err)
 	}
 
-	if _, ok := itemArr.(stackitem.Null); !ok {
-		arr, ok := itemArr.Value().([]stackitem.Item)
+	if _, ok := item.(stackitem.Null); !ok {
+		arr, ok := item.Value().([]stackitem.Item)
 		if !ok {
 			// unexpected for types from stackitem package
 			return cid.ID{}, errors.New("invalid cast to stack item slice")
 		}
 
 		var id cid.ID
-
```

Comment:
I'd complain about this in the overall diff. Not critical though.

Reply:
fixed

Reply:
fixed